blocks/filter.py

```diff
@@ -124,10 +140,17 @@ def __call__(self, variables):
         if self.name:
             variables = [var for var in variables
                          if hasattr(var.tag, 'name') and
+                         self.name == var.tag.name]
+        if self.name_re:
+            variables = [var for var in variables
+                         if hasattr(var.tag, 'name') and
                          re.match(self.name, var.tag.name)]
-        if self.application:
+        if self.applications:
             variables = [var for var in variables
                          if get_application_call(var) and
-                         get_application_call(var).application ==
-                         self.application]
-        return variables
+                         get_application_call(var).application in
+                         self.applications]
+        if self.one:
+            if len(variables) > 1:
+                raise ValueError("Variable Filter only return one variable")
```

Comment:
This error message isn't very grammatical, and a bit unclear. How about "variable filter matched multiple variables"?